theano/gof/vm.py

```diff
@@ -717,7 +717,13 @@ def make_vm(self, nodes, thunks,
                 updated_vars
                 ):
 
-        pre_call_clear = [storage_map[v] for v in self.no_recycling]
+        # Only put in pre_call clear variables that have not been computed
+        # inplace
+        pre_call_clear = [storage_map[v] for v in self.no_recycling
+                          if v.owner is None or
+                          not hasattr(v.owner.op, 'destroy_map') or
+                          v.owner.outputs.index(v) not in
+                          v.owner.op.destroy_map.keys()]
```

Comment:
I would rather you modify the place where no_recycling is created than hack at in in VM.  Also this will break the scan-no-copy if you use another linker.